File: DynamicScrollView/Component/ExampleView.m
//
//  LicentiaVenueView.m
//  DynamicScrollView
//
//  Created by Damir Stuhec on 29/06/14.
//  Copyright (c) 2014 damirstuhec. All rights reserved.
//

#import "ExampleView.h"

@implementation ExampleView

- (id)initWithFrame:(CGRect)frame name:(NSString *)name
{
    self = [super initWithFrame:frame];
    if (self) {
        CGFloat hue = ( arc4random() % 256 / 256.0 );
        CGFloat saturation = ( arc4random() % 128 / 256.0 ) + 0.5;
        CGFloat brightness = ( arc4random() % 128 / 256.0 ) + 0.5;
        
        self.backgroundColor = [UIColor colorWithHue:hue saturation:saturation brightness:brightness alpha:1];
        
        self.nameLabel = [[UILabel alloc] initWithFrame:frame];
        self.nameLabel.text = name;
        self.nameLabel.textAlignment = NSTextAlignmentCenter;
        self.nameLabel.autoresizingMask = UIViewAutoresizingFlexibleHeight|UIViewAutoresizingFlexibleWidth;
        
        [self addSubview:self.nameLabel];
    }
    return self;
}

- (BOOL)isEqual:(id)object
{
    ExampleView *anotherView = ([object isKindOfClass:[ExampleView class]]) ? (ExampleView *)object : nil;
    return self.uniqueNumber == anotherView.uniqueNumber;
}

- (NSComparisonResult)compareToObject:(id)object
{
    ExampleView *anotherView = ([object isKindOfClass:[ExampleView class]]) ? (ExampleView *)object : nil;
    
    if (self.distance > anotherView.distance)
    {
        return NSOrderedDescending;
    }
    else
    {
        return NSOrderedAscending;
    }
}

@end

File: Sample_Swift/DynamicScrollView/ExampleView.swift
//
//  ExampleView.swift
//  DynamicScrollView
//
//  Created by Damir Stuhec on 03/10/14.
//  Copyright (c) 2014 damirstuhec. All rights reserved.
//

import UIKit
import Darwin

class ExampleView: DSDynamicView
{
    var nameLabel: UILabel
    var uniqueNumber: Int
    var distance: Int
    
    init(frame: CGRect, name: String)
    {
        nameLabel = UILabel(frame: frame)
        nameLabel.autoresizingMask = .FlexibleHeight | .FlexibleWidth
        nameLabel.textAlignment = .Center
        nameLabel.text = name
        nameLabel.textColor = UIColor.whiteColor()
        
        uniqueNumber = 0
        distance = 0
        
        super.init(frame: frame)
        
        var hue = CGFloat(Float(arc4random() % 256) / 256.0)
        var saturation = CGFloat(Float(arc4random() % 128) / 256.0) + 0.5
        var brightness = CGFloat(Float(arc4random() % 128) / 256.0) + 0.5
        
        self.backgroundColor = UIColor(hue: CGFloat(hue), saturation: CGFloat(saturation), brightness: CGFloat(brightness), alpha: 1)
        self.addSubview(nameLabel)
    }
    
    override func isEqual(object: AnyObject?) -> Bool
    {
        if let anotherView = object as? ExampleView
        {
            return self.uniqueNumber == anotherView.uniqueNumber
        }
        else
        {
            return false
        }
    }
    
    override func compareToObject(object: AnyObject?) -> Bool
    {
        if let anotherView = object as? ExampleView
        {
            if self.distance > anotherView.distance
            {
                return false
            }
            else
            {
                return true
            }
        }
        else
        {
            return true
        }
    }
}
